SimPEG/regularization/regularization_mesh.py

```diff
@@ -551,8 +551,8 @@ def cell_distances_x(self) -> np.ndarray:
             Cell center distance array along the x-direction.
         """
         if getattr(self, "_cell_distances_x", None) is None:
-            Ave = self.aveCC2Fx
-            self._cell_distances_x = Ave * (self.Pac.T * self.mesh.h_gridded[:, 0])
+            self._cell_distances_x = self.cell_gradient_x.max(axis=1).data ** (-1.0)
```

Comment:
Sorry just one last change before this is ready (as I'm thinking about it).

It would likely be safer to instead use `.to_array()` instead of `.data`, That way you can be sure of the ordering of the output.

```suggestion
            self._cell_distances_x = self.cell_gradient_x.max(axis=1).toarray().flatten() ** (-1.0)
```

There are certain cases where the `.data` attribute wouldn't work out to being the correct ordering/ have the correct length, but `.toarray()` should work in all cases.